common/client.go

```diff
@@ -173,6 +175,9 @@ func (rc *RestClient) modifyUrl(dest string, queryMod url.Values) error {
 // execMethod applies the specified method to the provided url (which is interpreted
 // as relative or absolute).
 func (rc *RestClient) execMethod(method string, dest string, data interface{}, result interface{}) error {
+	// TODO check if token expired, if yes, reauthenticate... But this needs
```

Comment:
But the client doesn't need to know this, right? As long as the server it wishes to contact accepts the token it's all good. It's the job of the server to check that the token is still valid. If not, a 403 is returned and then the client will know to check again with auth to get a new token.